src/main/java/bio/terra/service/snapshot/flight/duos/RecordDuosFirecloudGroupStep.java

```diff
@@ -0,0 +1,41 @@
+package bio.terra.service.snapshot.flight.duos;
+
+import static bio.terra.service.snapshot.flight.duos.SnapshotDuosFlightUtils.getFirecloudGroup;
+
+import bio.terra.model.DuosFirecloudGroupModel;
+import bio.terra.service.duos.DuosDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class RecordDuosFirecloudGroupStep implements Step {
+
+  private final DuosDao duosDao;
+
+  public RecordDuosFirecloudGroupStep(DuosDao duosDao) {
+    this.duosDao = duosDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    UUID duosFirecloudGroupId = duosDao.insertFirecloudGroup(getFirecloudGroup(context));
```

Comment:
I would say typically we'd want to so the insert/retrieve in a service method and not call dao's directly from the step (so that you don't need to do the create and get.

Reply:
That way you run the create and get in a Xaction too

Reply:
Wrapping insert-and-retrieve in a service method sounds reasonable.

I'm not fully following your second comment though - "That way you run the create and get in a Xaction too", can you elaborate further?

Reply:
Ah, got it!  I heard "create" and my mind went to Sam resource creation which is handled in another step, this makes more sense.

Reply:
I ended up adding `DuosDao.insertAndCreateFirecloudGroup` method. We still need the DAO for undoing the step (deleting the record if we successfully wrote it), and it felt to me like a method that made sense for the DAO.  But if you would prefer it live in `DuosService`, let me know!